utils/src/main/java/com/cloud/utils/net/NetUtils.java

```diff
@@ -99,6 +99,10 @@ public class NetUtils {
     public final static int IPV6_EUI64_11TH_BYTE = -1;
     public final static int IPV6_EUI64_12TH_BYTE = -2;
 
+    // Regex
+    public final static Pattern HOSTNAME_PATTERN = Pattern.compile("[a-zA-Z0-9-]+");
+    public final static Pattern START_HOSTNAME_PATTERN = Pattern.compile("^[0-9-].*");
```

Comment:
would this work?

```suggestion
    public final static Pattern HOSTNAME_PATTERN = Pattern.compile("^(a-zA-Z)[a-zA-Z0-9-]+");
```

we could skip the START_HOSTNAME_PATTERN check if we check the hot name like this

Reply:
yes, can skip the start pattern for host name. for now, keeping it as separate check.